pkg/broker/ingress/ingress_handler.go

```diff
@@ -228,22 +228,9 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 	if features.IsOIDCAuthentication() {
 		h.Logger.Debug("OIDC authentication is enabled")
 
-		if broker.Status.Address.Audience == nil {
-			h.Logger.Warn(fmt.Sprintf("Audience of broker %s/%s must not be nil, while feature %s is enabled", broker.Name, broker.Namespace, feature.OIDCAuthentication))
-			writer.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		token := auth.GetJWTFromHeader(request.Header)
-		if token == "" {
-			h.Logger.Warn(fmt.Sprintf("No JWT in %s header provided while feature %s is enabled", auth.AuthHeaderKey, feature.OIDCAuthentication))
-			writer.WriteHeader(http.StatusUnauthorized)
-			return
-		}
-
-		if _, err := h.tokenVerifier.VerifyJWT(ctx, token, *broker.Status.Address.Audience); err != nil {
-			h.Logger.Warn("no valid JWT provided", zap.Error(err))
-			writer.WriteHeader(http.StatusUnauthorized)
+		writer, err = auth.VerifyJWTFromRequest(ctx, h.tokenVerifier, request, *broker.Status.Address.Audience, writer)
```

Comment:
passing the audience directly can lead to a nil pointer panic (you don't check it anymore like on line 231). As you check the audience in VerifyJWTFromRequest anyhow, you could pass a pointer here too :shrug: